egs/wsj/s5/steps/nnet3/nnet3_train_lib.py

```diff
@@ -175,25 +183,28 @@ def SplitData(data, num_jobs):
                                                                   num_jobs = num_jobs))
 
 def ParseModelConfigVarsFile(var_file):
+    variables = {}
     try:
         var_file_handle = open(var_file, 'r')
-        model_left_context = None
-        model_right_context = None
-        num_hidden_layers = None
         for line in var_file_handle:
             parts = line.split('=')
+            if len(parts) == 0:
+                continue
+            if len(parts) != 2:
+                break
             field_name = parts[0].strip()
-            field_value = parts[1]
+            field_value = parts[1].strip()
             if field_name in ['model_left_context', 'left_context']:
-                model_left_context = int(field_value)
+                variables['model_left_context'] = int(field_value)
             elif field_name in ['model_right_context', 'right_context']:
-                model_right_context = int(field_value)
+                variables['model_right_context'] = int(field_value)
             elif field_name == 'num_hidden_layers':
-                num_hidden_layers = int(field_value)
-
-        if model_left_context is not None and model_right_context is not None and num_hidden_layers is not None:
-            return [model_left_context, model_right_context, num_hidden_layers]
-
+                variables['num_hidden_layers'] = int(field_value)
+            elif field_name == 'add_lda':
```

Comment:
better do a more generic conversion. If field_value is true/false convert it to python bool